statsd/statsd.go

```diff
@@ -280,6 +287,15 @@ func NewWithWriter(w statsdWriter, options ...Option) (*Client, error) {
 	return newWithWriter(w, o, "custom")
```

Comment:
Maybe fields `options` and `addrOption` can also be set in `NewWithWriter`?

Reply:
The issue with with `NewWithWriter` is that both client would share the same writer. This means that Closing one client would close the transport of the other one too.
That's why we raise an error when cloning a client created with `NewWithWriter`.